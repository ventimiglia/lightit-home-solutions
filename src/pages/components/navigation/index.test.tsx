import { render } from "@testing-library/react";
import Navigation from "./";

describe("Navigation", () => {
  test("renders buttons with correct labels", () => {
    const { container, getByText, getByAltText } = render(
    <Navigation>
      <Navigation.Buttons />
      <Navigation.Controllers />
    </Navigation>);

    const fijarButton = getByText("Fijar");
    const borrarButton = getByText("Borrar");
    const aumentarButton = getByAltText("plus");
    const disminuirButton = getByAltText("less");
    const arribaButton = getByAltText("up");
    const abajoButton = getByAltText("down");
    const izquierdaButton = getByAltText("left");
    const derechaButton = getByAltText("right");

    expect(fijarButton).toBeInTheDocument();
    expect(borrarButton).toBeInTheDocument();
    expect(aumentarButton).toBeInTheDocument();
    expect(disminuirButton).toBeInTheDocument();
    expect(arribaButton).toBeInTheDocument();
    expect(abajoButton).toBeInTheDocument();
    expect(izquierdaButton).toBeInTheDocument();
    expect(derechaButton).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
