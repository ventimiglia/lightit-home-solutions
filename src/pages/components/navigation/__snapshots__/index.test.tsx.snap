// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders buttons with correct labels 1`] = `
<div>
  <aside
    class="flex flex-col justify-between flex-1 items-end p-4 relative"
  >
    <div
      class="flex gap-4"
    >
      <button
        class="bg-white w-24 py-3 rounded-lg text-center text-primary transition duration-300 ease-in-out hover:bg-primary hover:text-white transform hover:scale-105 focus:outline-none focus:ring-2 focus:ring-primary"
      >
        Fijar
      </button>
      <button
        class="bg-white w-24 py-3 rounded-lg text-center text-primary transition duration-300 ease-in-out hover:bg-primary hover:text-white transform hover:scale-105 focus:outline-none focus:ring-2 focus:ring-primary"
      >
        Borrar
      </button>
    </div>
    <div
      class="flex gap-4 absolute bottom-4"
    >
      <div
        class="flex flex-col flex-wrap justify-between bg-white rounded-md p-1"
      >
        <button
          class="transition duration-300 ease-in-out transform hover:scale-125"
        >
          <img
            alt="plus"
            src="plus.png"
            width="22"
          />
        </button>
        <hr />
        <button
          class="transition duration-300 ease-in-out transform hover:scale-125"
        >
          <img
            alt="less"
            src="less.png"
            width="22"
          />
        </button>
      </div>
      <div
        class="w-16 h-16 relative bg-white border-4 border-white rounded-md"
      >
        <button
          class="w-5 h-5 absolute top-0 right-1/2 translate-x-1/2 transition duration-300 ease-in-out transform hover:scale-125"
          name="up"
        >
          <img
            alt="up"
            src="arrow.png"
          />
        </button>
        <button
          class="w-5 h-5 absolute right-0 top-1/2 -translate-y-1/2 transition duration-300 ease-in-out transform hover:scale-125"
          name="right"
        >
          <img
            alt="right"
            class="rotate-90"
            src="arrow.png"
          />
        </button>
        <button
          class="w-5 h-5 absolute left-0 top-1/2 -translate-y-1/2 transition duration-300 ease-in-out transform hover:scale-125"
          name="left"
        >
          <img
            alt="left"
            class="-rotate-90"
            src="arrow.png"
          />
        </button>
        <button
          class="w-5 h-5 absolute bottom-0 left-1/2 -translate-x-1/2 transition duration-300 ease-in-out transform hover:scale-125"
          name="down"
        >
          <img
            alt="down"
            class="rotate-180"
            src="arrow.png"
          />
        </button>
      </div>
    </div>
  </aside>
</div>
`;
